csrc/device_lower/pass/circular_buffer.cpp

```diff
@@ -387,11 +961,51 @@ class CircularBufferInserter : private kir::ExprMutator {
       return;
     }
 
-    insert(loop, it->second);
+    auto hasCpAsyncBulk = std::any_of(
+        it->second.begin(), it->second.end(), ir_utils::isCpAsyncBulk);
+
+    if (hasCpAsyncBulk) {
+      insertTma(loop, it->second);
+    } else {
+      insert(loop, it->second);
+    }
     processed_loop_ = loop;
     insertion_info_.erase(loop);
   }
 
+  void insertTma(
+      ForLoop* circular_buffer_loop,
+      const std::vector<Expr*>& loads) {
+    // Prologue loop:
+    //  - launch only
+    //  - arrive_expect_tx and tma load operations
+    ForLoop* prologue_loop = TmaCircularBufferLoopCloner::clone(
```

Comment:
Should we rename `TmaCircularBufferLoopCloner` as `CloneTmaCircularBufferLoopAndInsertSync`? Compared with the traditional `CircularBufferLoopCloner`, which only clone expressions, `TmaCircularBufferLoopCloner` also insert syncs. Failing to realize this could make it hard to understand the logic in code.